### PR TITLE
Improve unpacking of solutions

### LINKED ISSUE #2: python checker finds file, but running the tests fails
Here is a link to an example: /prog2-aki/admin/solutions/solution/658/

I guess that the problem is the following: 

* the student zipped the solution such that the .py files reside in their own directory
* the python checkers finds the files
* but running the tests does not take the subdirectory into account and thus fails

This is a pretty bad problem because in the UI, all files seem present (without subdirectory), but the tests are failing.

@hannesbraun can you have a look? After submission of your thesis, of course.

### PR DESCRIPTION
**Pull request by hannesbraun**
In the case that students wrap their solutions with unnecessary folders, the multi-checker will now try to determine the actual directory that contains the solution.

Fixes #2

### CODE REVIEW

File: multi-checker/script/utils.py [hunk 1]
```diff
@@ -82,3 +82,31 @@ def isDebug():
...
     if _DEBUG:
         print(f'[DEBUG] {msg}')
+
+def findDirMatchingAux(root: str, stopCondition: Callable) -> Optional[str]:
+    if isFile(root):
+        return None
+    if stopCondition(root):
+        return root
+    for x in ls(root):
+        result = findDirMatchingAux(x, stopCondition)
```
Comment (skogsbaer) [inline multi-checker/script/utils.py]: only descend recursively if root is a wrapper dir. For example, you do not want to descend in the following case:

root/A/src
root/B/src

File: multi-checker/script/utils.py [hunk 2]
```diff
@@ -82,3 +82,31 @@ def isDebug():
...
+
+def isNotAWrapperDir(directory: str) -> bool:
+    pathnames = ls(directory, '[!.]*')
+    if len(pathnames) > 1:
+        return True
+    if len(pathnames) == 0:
+        return False
+    return not isDir(pathnames[0])
+
+def findDirMatching(root: str, stopCondition: Callable) -> str:
```
Comment (skogsbaer) [inline multi-checker/script/utils.py]: Short documentation

**Review by skogsbaer — CHANGES_REQUESTED** [review]
I think a couple of test cases would be good.

File: multi-checker/script/utils.py (hunk 1 above)
Comment (hannesbraun) [reply multi-checker/script/utils.py]: This should be fixed with 8140d9dcbca827ba3c592f320c5ead875b561e00.

File: multi-checker/script/utils.py (hunk 2 above)
Comment (hannesbraun) [reply multi-checker/script/utils.py]: Done

**Pull request merged**